### PR TITLE
refactor: update `blas/ext/base/dapxsumors` to follow current project conventions 

### LINKED ISSUE #1468: [RFC]: refactor `blas/ext/base/dapxsumors` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dapxsumors`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dapxsumors) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by ShivamAhir**
Resolves # .

## Description
 update blas/ext/base/dapxsumors to follow current project conventions #1468

This pull request:

-   a
-   b
-   c

## Related Issues

> Does this pull request have any related issues? 

This pull request:

-   resolves #1468 
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumors/README.md [hunk 1]
```diff
@@ -139,18 +133,11 @@ var v = dapxsumors.ndarray( N, 5.0, x, 2, 1 );
...
 var dapxsumors = require( '@stdlib/blas/ext/base/dapxsumors' );
 
-var x;
-var i;
-
-x = new Float64Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-    x[ i ] = round( randu()*100.0 );
-}
+var x = filledarrayBy(10, 'float32', discreteUniform(0, 100));
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumors/README.md]:
```suggestion
var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumors/docs/repl.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@
     Adds a constant to each double-precision floating-point strided array
     element and computes the sum using ordinary recursive summation.
 
-    The `N` and `stride` parameters determine which elements in `x` are accessed
+    The `N` and stride parameters determine which elements in the strided array are accessed
     at runtime.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumors/docs/repl.txt]:
```suggestion
    The `N` and stride parameters determine which elements in the strided array are 
    accessed at runtime.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumors/manifest.json [hunk 3]
```diff
@@ -34,7 +34,12 @@
...
 			"libpath": [],
-			"dependencies": []
+			"dependencies": [
+				"stdlib/napi/export.h",
+				"stdlib/napi/argv.h",
+				"stdlib/napi/argv_int64.h",
+      			"stdlib/napi/argv_strided_float64array.h"
+			]
 		}
 	]
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumors/manifest.json]: The indentation in the `manifest.json` file should be changed to 2-spaces and configurations for the examples and benchmarks should be added, see the linked PRs in the RFC.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumors/src/addon.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+    STDLIB_NAPI_ARGV_DOUBLE( env, alpha, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, stride, argv, 2 );
+
+	napi_value v;
+	napi_status status = napi_create_double( env, stdlib_strided_dapxsumors( N, alpha, (double *)X, stride ), &v );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumors/src/addon.c]: Casting to double is not necessary anymore here:

```suggestion
	napi_status status = napi_create_double( env, stdlib_strided_dapxsumors( N, alpha, X, stride ), &v );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumors/test/test.dapxsumors.js [hunk 5]
```diff
@@ -110,8 +109,7 @@ tape( 'the function supports a `stride` parameter', function test( t ) {
 		2.0
 	]);
 
-	N = floor( x.length / 2 );
-	v = dapxsumors( N, 5.0, x, 2 );
+	v = dapxsumors( 4, 5.0, x, 2 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumors/test/test.dapxsumors.js:111]: Please remove `N` variable declarations in this and the other test files as they are now unused.

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks for working on this! Left a bunch of requested changes that should hopefully be easy to resolve.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumors/benchmark/benchmark.js [hunk 6]
```diff
@@ -21,13 +21,17 @@
...
 
 var bench = require( '@stdlib/bench' );
-var randu = require( '@stdlib/random/base/randu' );
+var uniform = require( '@stdlib/random/base/uniform' ).factory;
+var filledarrayBy = require( '@stdlib/array/filled-by' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float64Array = require( '@stdlib/array/float64' );
 var pkg = require( './../package.json' ).name;
 var dapxsumors = require( './../lib/dapxsumors.js' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumors/benchmark/benchmark.js:30]:
```suggestion


```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumors/benchmark/benchmark.ndarray.js [hunk 7]
```diff
@@ -21,13 +21,17 @@
...
 
 var bench = require( '@stdlib/bench' );
-var randu = require( '@stdlib/random/base/randu' );
+var uniform = require( '@stdlib/random/base/uniform' ).factory;
+var filledarrayBy = require( '@stdlib/array/filled-by' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float64Array = require( '@stdlib/array/float64' );
 var pkg = require( './../package.json' ).name;
 var dapxsumors = require( './../lib/ndarray.js' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumors/benchmark/benchmark.ndarray.js:30]:
```suggestion


```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumors/docs/repl.txt [hunk 8]
```diff
@@ -39,18 +39,15 @@
...
 
     // Using view offsets:
     > var x0 = new {{alias:@stdlib/array/float64}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ] );
     > var x1 = new {{alias:@stdlib/array/float64}}( x0.buffer, x0.BYTES_PER_ELEMENT*1 );
-    > N = {{alias:@stdlib/math/base/special/floor}}( x0.length / 2 );
-    > stride = 2;
-    > {{alias}}( N, 5.0, x1, stride )
-    14.0
+    > {{alias}}( 2, 5.0, x1, 2 )
+    10
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumors/docs/repl.txt]:
```suggestion
    > {{alias}}( 3, 5.0, x1, 2 )
    14.0
```

**Review by Planeshifter — APPROVED** [review]
Thanks! Will merge shortly after CI has passed.

**Comment by ShivamAhir** [comment]
@Planeshifter grateful for your ongoing guidance and support. Hard to manage without it.😊

**Pull request merged**